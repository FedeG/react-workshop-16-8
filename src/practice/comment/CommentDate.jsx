import React from 'react';
import PropTypes from 'prop-types';

const formatDate = date => date.toDateString();

const CommentDate = ({ date }) => (
  <div className="Comment-date">
    {formatDate(date)}
  </div>
);

CommentDate.propTypes = {
  date: PropTypes.instanceOf(Date).isRequired,
};

export default CommentDate;
