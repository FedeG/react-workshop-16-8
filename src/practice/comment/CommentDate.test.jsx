import React from 'react';
import MockDate from 'mockdate';
import renderer from 'react-test-renderer';

import CommentDate from './CommentDate';

describe('CommentDate', () => {
  let component;

  beforeEach(() => {
    MockDate.set('3/5/1995');
    component = renderer.create(
      <CommentDate date={new Date()} />,
    );
  });

  test('initial state', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
