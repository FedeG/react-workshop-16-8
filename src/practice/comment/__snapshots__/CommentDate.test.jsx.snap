// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentDate initial state 1`] = `
<div
  className="Comment-date"
>
  Sun Mar 05 1995
</div>
`;
